js/Puck.js

```diff
@@ -1,15 +1,30 @@
 class Puck {
     #puckElement;
-    #rowLengths;
+    #rowLengths = [];
     #selectedCellIndexes;
     #onCellClickCb = [];
     #onCellHoverCb = [];
-    constructor(puckElement,rowLengths, selectedCellIndexes){
+    constructor(puckElement, rowLengths, selectedCellIndexes){
         this.#puckElement = puckElement;
-        this.#rowLengths =rowLengths;
         this.#selectedCellIndexes = selectedCellIndexes;
+        this.#rowLengths = rowLengths;
+    }
+    changeRowLengths(){
```

Comment:
In real life we will never change the dimensions of a puck. We need this only for the demo page, not for real projects. So it would be better to instead delete and redraw the whole puck one a button is pressed.